File: src/commands/remove.ts
import inquirer from 'inquirer';
import chalk from 'chalk';
import { getWorktrees, removeWorktree } from '../utils.js';

export async function removeWorktreeCommand(): Promise<void> {
  try {
    console.log(chalk.blue('🗑️  Removing a Git worktree\n'));

    const worktrees = await getWorktrees();

    if (worktrees.length === 0) {
      console.log(chalk.yellow('No worktrees found to remove.'));
      return;
    }

    // Create choices for inquirer
    const choices = worktrees.map(wt => ({
      name: `${wt.branch} (${wt.path})`,
      value: wt.path
    }));

    const { selectedWorktree } = await inquirer.prompt([
      {
        type: 'list',
        name: 'selectedWorktree',
        message: 'Select a worktree to remove:',
        choices
      }
    ]);

    // Confirm deletion
    const { confirmDelete } = await inquirer.prompt([
      {
        type: 'confirm',
        name: 'confirmDelete',
        message: `Are you sure you want to remove this worktree?\n  ${selectedWorktree}`,
        default: false
      }
    ]);

    if (!confirmDelete) {
      console.log(chalk.yellow('Operation cancelled.'));
      return;
    }

    console.log(chalk.cyan(`\n🗑️  Removing worktree: ${selectedWorktree}`));

    // Remove the worktree
    await removeWorktree(selectedWorktree);
    
    console.log(chalk.green('✅ Worktree removed successfully!'));
    
  } catch (error) {
    console.error(chalk.red(`❌ Error: ${error}`));
    process.exit(1);
  }
}

File: src/commands/remove.rs
use clap::Args;
use anyhow::Result;
use crate::git::repository::Repository;
use crate::git::worktree::WorktreeManager;
use crate::ui::prompts::Prompts;
use crate::ui::display::Display;

#[derive(Debug, Args)]
pub struct RemoveCommand {}

impl RemoveCommand {
    pub fn execute(&self) -> Result<()> {
        Display::show_info("Removing a worktree...");
        
        let repo = Repository::open_current()?;
        let worktree_manager = WorktreeManager::new(repo.inner.path().parent().unwrap().to_path_buf());
        
        // 獲取 worktree 列表
        let worktrees = worktree_manager.list_worktrees()?;
        
        if worktrees.is_empty() {
            Display::show_info("No worktrees found to remove.");
            return Ok(());
        }
        
        // 選擇要移除的 worktree
        let selected_worktree = Prompts::select_worktree(worktrees)?;
        
        // 確認移除
        if Prompts::confirm_removal(&selected_worktree)? {
            Display::show_info("Removing worktree...");
            worktree_manager.remove_worktree(&selected_worktree.path)?;
            Display::show_success(&format!("Worktree '{}' removed successfully!", selected_worktree.branch));
        } else {
            Display::show_info("Operation cancelled.");
        }
        
        Ok(())
    }
}